tests/check_illegal_windows_names_test.py

```diff
@@ -0,0 +1,78 @@
+import sys
+
+import pytest
+
+from pre_commit_hooks.check_illegal_windows_names import (
+    find_illegal_windows_names,
+)
+from pre_commit_hooks.check_illegal_windows_names import main
+from pre_commit_hooks.check_illegal_windows_names import parents
+from pre_commit_hooks.util import cmd_output
+
+skip_win32 = pytest.mark.skipif(
+    sys.platform == 'win32',
+    reason='case conflicts between directories and files',
+)
```

Comment:
I think once you remove `added_files` as a constraint you won't need to skip anything (you won't even need to have any disk state either!)